modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/GcalController.scala

```diff
@@ -0,0 +1,87 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.spModel.gemini.calunit.CalUnitParams.Shutter
+
+/**
+  * Created by jluhrs on 3/15/17.
+  */
+trait GcalController {
+
+  import GcalController._
+
+  def getConfig: SeqAction[GcalConfig]
+
+  def applyConfig(config: GcalConfig): SeqAction[Unit]
+
+}
+
+object GcalController {
+
+  trait LampState
+
+  object LampState {
+
+    case object Off extends LampState
+
+    case object On extends LampState
+
+  }
+
+  final case class ArLampState(self: LampState) extends AnyVal
+
+  final case class CuArLampState(self: LampState) extends AnyVal
+
+  final case class QHLampState(self: LampState) extends AnyVal
+
+  final case class ThArLampState(self: LampState) extends AnyVal
+
+  final case class XeLampState(self: LampState) extends AnyVal
+
+  final case class IrLampState(self: LampState) extends AnyVal
```

Comment:
Those classes are to avoid confusion when building a configuration. They make it harder to swap them by mistake.